src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/AbstractPathJob.java

```diff
@@ -1341,4 +1398,14 @@ public void setPathingOptions(final PathingOptions pathingOptions)
         DROPABLE,
         NOT_PASSABLE
     }
+
+    /**
+     * Check if in restricted area.
+     * @param pos the pos to check.
+     * @return true if so.
+     */
+    public boolean isInRestrictedArea(final BlockPos pos)
```

Comment:
could even make this a general canWalkNode(pos) which just defaults to true and specialized jobs can override and then we dont need any restriction logic at the highest level which is only seeing use in the special cases anyways

Reply:
Well, in this use-case it's a specific pass function which we already override. The method itself is just a utility method we can call.